src/librustc_trans/glue.rs

```diff
@@ -205,11 +210,10 @@ pub fn implement_drop_glue<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>, g: DropGlueKi
     };
 
     let bcx = match t.sty {
-        ty::TyBox(content_ty) => {
-            // Support for TyBox is built-in and its drop glue is
-            // special. It may move to library and have Drop impl. As
-            // a safe-guard, assert TyBox not used with TyContents.
-            assert!(!skip_dtor);
```

Comment:
It might be not used at all. @michaelwoerister and @arielb1 would know.
What I meant is making the `trans_exchange_free_ty` call conditional on `!skip_dtor`.